clients/unshaded/src/main/java/tachyon/client/file/FileInStream.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client.file;
+
+import java.io.IOException;
+import java.util.List;
+
+import com.google.common.base.Preconditions;
+
+import tachyon.client.ClientOptions;
+import tachyon.client.InStream;
+import tachyon.client.block.BlockInStream;
+import tachyon.client.block.BlockOutStream;
+import tachyon.client.block.LocalBlockInStream;
+import tachyon.master.block.BlockId;
+import tachyon.thrift.FileInfo;
+
+/**
+ * Provides a streaming API to read a file. This class wraps the BlockInStreams for each of the
+ * blocks in the file and abstracts the switching between streams. The backing streams can read from
+ * Tachyon space in the local machine, remote machines, or the under storage system.
+ */
+public final class FileInStream extends InStream {
+  private final boolean mShouldCache;
+  private final long mBlockSize;
+  private final long mFileLength;
+  private final FSContext mContext;
+  private final List<Long> mBlockIds;
+  private final String mUfsPath;
+  private final ClientOptions mOptions;
+
+  private boolean mClosed;
+  private boolean mShouldCacheCurrentBlock;
+  private long mPos;
+  private BlockInStream mCurrentBlockInStream;
+  private BlockOutStream mCurrentCacheStream;
+
+  public FileInStream(FileInfo info, ClientOptions options) {
```

Comment:
maybe  have an overloading constructor takes no `ClientOptions` which assumes `ClientOptions::defaults()` is used?